streaming/base/dataset.py

```diff
@@ -202,12 +205,35 @@ def __init__(self,
         # Determine and distribute shuffle seed and shm prefix.
         seed_rng = np.random.default_rng(shuffle_seed)
         self.shuffle_seed = int(seed_rng.integers(1 << 60))
-        prefix_int = np.random.randint(1 << 24)
+        prefix_int = int(seed_rng.integers(1 << 24))
         self._prefix = f'{prefix_int:06x}'
 
         # Should be a unique shared directory per each StreamingDataset instantiation to avoid a conflict
         # between a different StreamingDataset instance on a same machine.
-        self._shared_dir = os.path.join(os.path.sep, 'tmp', 'streaming', self._prefix)
+        start_time = time()
+        while True:
+            sleep(TICK)
+            self._shared_dir = os.path.join(os.path.sep, 'tmp', 'streaming', self._prefix)
+            if os.path.exists(self._shared_dir):
+                prefix_int = int(seed_rng.integers(1 << 24))
+                self._prefix = f'{prefix_int:06x}'
+            else:
+                break
+            dt = time() - start_time
+            if dt > TIMEOUT:
+                raise RuntimeError(
+                    f'Could not find the unique shared directory, bailing out. Please provide a different `shuffle_seed` value.'
```

Comment:
100 col

Reply:
Fixed in https://github.com/mosaicml/streaming/pull/147/commits/c231d56705f8d8030f7b2b9dd532040ef6dceafc.